src/data_structure/mga_data.jl

```diff
@@ -0,0 +1,219 @@
+            for ind in mga_indices(mga_current_iteration)
+                d_aux[ind] = @variable(m, base_name = _base_name(:mga_aux_diff,ind), lower_bound = 0)
+                d_bin[ind] = @variable(m, base_name = _base_name(:mga_aux_binary,ind), binary=true)
+            end
+            @fetch mga_aux_diff, mga_aux_binary, mga_objective = m.ext[:variables]
+            mga_results = m.ext[:outputs]
+            variable = m.ext[:variables][variable_name]
+            #FIXME: don't create new dict everytime, but get existing one
+            d_diff_ub1 = get!(m.ext[:constraints],:mga_diff_ub1,Dict())
+            d_diff_ub2 = get!(m.ext[:constraints],:mga_diff_ub2,Dict())
+            d_diff_lb1 = get!(m.ext[:constraints],:mga_diff_lb1,Dict())
+            d_diff_lb2 = get!(m.ext[:constraints],:mga_diff_lb2,Dict())
+            for ind in mga_indices()
+                d_diff_ub1[(ind...,mga_current_iteration...)] = @constraint(
+                    m,
+                    mga_aux_diff[((ind...,mga_iteration=mga_current_iteration))]
+                    <=
+                    sum(
+                    + (
+                    variable[_ind]
+                     - mga_results[variable_name][((Base.structdiff(_ind,NamedTuple{(:t,)}(_ind.t))..., mga_iteration=mga_current_iteration))][t0.ref.x][_ind.t.start.x]
+                     )
+                     *scenario_weight_function(m; Base.structdiff(_ind,NamedTuple{(:t,)}(_ind.t))...) #fix me, can also be only node or so
+                           for _ind in variable_indices_function(m; ind...)
+                   )
+                   + mga_variable_bigM(;ind...)*mga_aux_binary[(ind...,mga_iteration=mga_current_iteration)])
+                d_diff_ub2[(ind...,mga_current_iteration...)]= @constraint(
+                    m,
+                    mga_aux_diff[((ind...,mga_iteration=mga_current_iteration))]
+                    <=
+                    sum(
+                    - (variable[_ind]
+                      - mga_results[variable_name][((Base.structdiff(_ind,NamedTuple{(:t,)}(_ind.t))..., mga_iteration=mga_current_iteration))][t0.ref.x][_ind.t.start.x])
+                      * scenario_weight_function(m; Base.structdiff(_ind,NamedTuple{(:t,)}(_ind.t))...)
+                           for _ind in variable_indices_function(m; ind...)
+                   )
+                  + mga_variable_bigM(;ind...)*(1-mga_aux_binary[(ind...,mga_iteration=mga_current_iteration)])
+                  )
+                  d_diff_lb1[(ind...,mga_current_iteration...)] = @constraint(
+                    m,
+                    mga_aux_diff[((ind...,mga_iteration=mga_current_iteration))]
+                    >=
+                    sum(
+                    (variable[_ind]
+                      - mga_results[variable_name][((Base.structdiff(_ind,NamedTuple{(:t,)}(_ind.t))..., mga_iteration=mga_current_iteration))][t0.ref.x][_ind.t.start.x])

```

Comment:
The call to Base.strucdiff is pretty difficult to read here. Is it just dropping a namedtuple key? Don't we have a function called drop_key somewhere to do that?